xds/internal/xdsclient/authority.go

```diff
@@ -86,89 +98,132 @@ type authority struct {
 	//
 	// The second level map key is the resource name, with the value being the
 	// actual state of the resource.
-	resourcesMu sync.Mutex
-	resources   map[xdsresource.Type]map[string]*resourceState
-	closed      bool
+	resources map[xdsresource.Type]map[string]*resourceState
+
+	// An ordered list of xdsChannels corresponding to the list of server
+	// configurations specified for this authority in the bootstrap. The
+	// ordering specifies the order in which these channels are preferred for
+	// fallback.
+	xdsChannelConfigs []*xdsChannelWithConfig
+
+	// The current active xdsChannel. Here, active does not mean that the
+	// channel has a working connection to the server. It simply points to the
+	// channel that we are trying to work with, based on fallback logic.
+	activeXDSChannel *xdsChannelWithConfig
 }
 
 // authorityArgs is a convenience struct to wrap arguments required to create a
-// new authority. All fields here correspond directly to appropriate fields
-// stored in the authority struct.
+// new authority.
 type authorityArgs struct {
-	// The reason for passing server config and bootstrap config separately
-	// (although the former is part of the latter) is because authorities in the
-	// bootstrap config might contain an empty server config, and in this case,
-	// the top-level server config is to be used.
-	serverCfg          *bootstrap.ServerConfig
-	bootstrapCfg       *bootstrap.Config
-	serializer         *grpcsync.CallbackSerializer
-	resourceTypeGetter func(string) xdsresource.Type
-	watchExpiryTimeout time.Duration
-	backoff            func(int) time.Duration // Backoff for ADS and LRS stream failures.
-	logger             *grpclog.PrefixLogger
+	serverConfigs    bootstrap.ServerConfigs      // Server configs for the authority
+	name             string                       // Name of the authority
+	serializer       *grpcsync.CallbackSerializer // Callback serializer for invoking watch callbacks
```

Comment:
Optional: call this watchCallbackSerializer to distinguish it from the local variable authority.serializer.